src/components/divider-dropdown-autocomplete/DividerDropdownAutocomplete.tsx

```diff
@@ -0,0 +1,65 @@
+import { SyntheticEvent } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AutocompleteRenderGroupParams } from '@mui/material'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import Divider from '@mui/material/Divider'
+
+import AsyncAutocomplete, {
+  AsyncAutocompleteProps
+} from '~/components/async-autocomlete/AsyncAutocomplete'
+
+import { styles } from '~/containers/my-courses/course-toolbar/CourseToolbar.style'
+
+interface DividedDropdownAutocompleteProps<
+  Response,
+  TransformedResponse,
+  Params
+> extends Omit<
+    AsyncAutocompleteProps<
+      Response,
+      Params,
+      TransformedResponse,
+      boolean | undefined
+    >,
+    'onChange'
+  > {
+  onChange: (_: SyntheticEvent, value: TransformedResponse | null) => void
+}
+
+const DividedDropdownAutocomplete = <
+  Response,
+  TransformedResponse,
+  Params = undefined
+>({
+  axiosProps,
+  groupBy,
+  ...props
+}: DividedDropdownAutocompleteProps<Response, TransformedResponse, Params>) => {
+  const { t } = useTranslation()
+
+  const optionsList = (params: AutocompleteRenderGroupParams) => {
```

Comment:
I think you can use destructuring for items of `params`

Reply:
I already destructured it below, or, do you mean to destructure it directly in params?

Reply:
> I already destructured it below or do you mean to destructure it directly in params?

in arguments of function like `( { }: AutocompleteRenderGroupParams )`

Reply:
changed